packages/lib/file_ops.js

```diff
@@ -69,7 +69,29 @@ async function findUnusedName(directory, name, extension = "") {
 	}
 }
 
+/**
+ * Safely write data to a file
+ *
+ * Same as fs-extra.outputFile except the data is written to a temporary
+ * file that's renamed over the target file.  The name of the temporary file
+ * is the same as the target file with the suffix `.tmp` added.
+ *
+ * If the operation fails it may leave behind the temporary file.  This
+ * should not be too much of an issue as the next time the same file is
+ * written the temporary will be overwritten and renamed to the target file.
+ *
+ * @param {string} file - Path to file to write.
+ * @param {string|Buffer} data - Content to write.
+ * @param {object|string} options - see fs.writeFile, `flag` must not be set.
+ */
+async function safeOutputFile(file, data, options={}) {
```

Comment:
If the temp file already exists, will this overwrite it or will it error out?
How about handling errors more gracefully? Cause you might be able to write the temp file, but not replace/overwrite the existing with the rename?

Reply:
If the temp file already exists its content is discarded (truncated) before the new data is written to it.

I'm not sure what graceful handling of errors there is to do. If the rename fails then the operation as a whole failed an it must throw to prevent the calling code from continuing without having written the data it expected.